libraries/chain/controller.cpp

```diff
@@ -4307,7 +4315,8 @@ struct controller_impl {
                     const flat_set<digest_type>& cur_features,
                     const vector<digest_type>& new_features )
             { check_protocol_features( timestamp, cur_features, new_features ); },
-            skip_validate_signee
+            skip_validate_signee,
+            check_canonical()
```

Comment:
Can't do this here. This is off the main thread. `is_builtin_activated` is not thread safe.

Reply:
https://github.com/AntelopeIO/spring/pull/1708/commits/74ac3486d588cdc62211abbb3eab839b1b5f5c0f

great that the tests failed because of this.